src/test/java/de/tum/in/www1/artemis/GradeStepIntegrationTest.java

```diff
@@ -107,10 +107,7 @@ public void testGetAllGradeStepsForCourse() throws Exception {
         assertThat(gradeStepsDTO.gradeType).isEqualTo(GradeType.GRADE);
         assertThat(gradeStepsDTO.title).isEqualTo(course.getTitle());
 
-        GradeStep[] gradeStepArray = new GradeStep[gradeSteps.size()];
-        gradeSteps.toArray(gradeStepArray);
-
-        assertThat(gradeStepsDTO.gradeSteps).usingRecursiveComparison().ignoringFields("gradingScale", "id").ignoringCollectionOrder().isEqualTo(gradeStepArray);
+        assertThat(gradeStepsDTO.gradeSteps).usingRecursiveComparison().ignoringFields("gradingScale", "id").ignoringCollectionOrder().isEqualTo(gradeStepsDTO.gradeSteps);
```

Comment:
```suggestion
        assertThat(gradeStepsDTO.gradeSteps).usingRecursiveComparison().ignoringFields("gradingScale", "id").ignoringCollectionOrder().isEqualTo(gradeSteps);
```
Otherwise you're comparing the same things! (Both times `gradeStepsDTO.gradeSteps`.)

Reply:
Good catch!